src/components/ResoucesList.tsx

```diff
@@ -0,0 +1,242 @@
+  } else {
+    returnText = text;
+  }
+  return returnText;
+}
+
+export function ResourcesList(): JSX.Element {
+  const { resources, updateResources } = useFetchResources();
+  const [searchValue, setSearchValue] = useState("");
+  const [filterTags, setFilterTags] = useState("");
+
+  const handleTagsFilter = (selectedValue: string) => {
+    const tagsArray = filterTags.split(", ");
+    if (tagsArray.includes(selectedValue)) {
+      const filteredTags = tagsArray.filter((tag) => tag !== selectedValue);
+      const arrayToString = filteredTags.join(", ");
+      return arrayToString;
+    }
+    if (filterTags.length === 0) {
+      return selectedValue;
+    }
+    if (filterTags.includes(selectedValue)) {
+      return "";
+    }
+    return filterTags + ", " + selectedValue;
+  };
+
+  const handleLikeResource = async (id: number) => {
+    try {
+      await axios.patch(BackendURL + `resources/likes/${id}`);
+      updateResources();
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  return (
+    <>
+      <div className="tags">
+        <div className="type-tags">
+          <label>TypeScript</label>
+          <input
+            value="TypeScript"
+            type="checkbox"
+            name="resource_tags"
+            onChange={(e) => setFilterTags(handleTagsFilter(e.target.value))}
+            required
+            className="resourceTag-TS"
+          ></input>
+          <label>React</label>
+          <input
+            value="React"
+            type="checkbox"
+            name="resource_tags"
+            onChange={(e) => setFilterTags(handleTagsFilter(e.target.value))}
+            className="resourceTag-React"
+            required
+          ></input>
+          <label>APIs</label>
+          <input
+            value="APIs"
+            type="checkbox"
+            name="resource_tags"
+            onChange={(e) => setFilterTags(handleTagsFilter(e.target.value))}
+            className="resourceTag-APIs"
+            required
+          ></input>
+          <label>Node</label>
+          <input
+            value="Node"
+            type="checkbox"
+            name="resource_tags"
+            onChange={(e) => setFilterTags(handleTagsFilter(e.target.value))}
+            required
+            className="resourceTag-Node"
+          ></input>
+          <label>JavaScript</label>
+          <input
+            value="JavaScript"
+            type="checkbox"
+            name="resource_tags"
+            onChange={(e) => setFilterTags(handleTagsFilter(e.target.value))}
+            required
+            className="resourceTag-JavaScript"
+          ></input>
+          <label>CSS</label>
+          <input
+            value="CSS"
+            type="checkbox"
+            name="resource_tags"
+            onChange={(e) => setFilterTags(handleTagsFilter(e.target.value))}
+            required
+            className="resourceTag-CSS"
+          ></input>
```

Comment:
I would save the tags in an array and then map over them following DRY principles (Do not repeat yourself), i.e. you would use the input structure just once.

Reply:
Thank you and yes you are absolutely correct, this was done in a hurry last minute. using an array is 100% better than what is done here